src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth.cpp

```diff
@@ -253,13 +308,59 @@ Unique<Earth::Impl> Earth::ImplFromType(
     }
     else if (aType == Earth::Type::Exponential)
     {
-        return std::make_unique<ExponentialImpl>(aType, anEarthFrameSPtr, anEarthRadius, anEarthFlattening);
+        return std::make_unique<ExponentialImpl>(
+            aType, anInputDataSourceType, anEarthFrameSPtr, anEarthRadius, anEarthFlattening
+        );
     }
     else if (aType == Earth::Type::NRLMSISE00)
     {
-        return std::make_unique<NRLMSISE00Impl>(
-            aType, anEarthFrameSPtr, anEarthRadius, anEarthFlattening, aSunCelestialSPtr
-        );
+        if (anInputDataSourceType == Earth::InputDataSourceType::Undefined)
+        {
+            throw ostk::core::error::runtime::Undefined("Input data source type");
```

Comment:
```suggestion
            Earth::InputDataSourceType defaultNRLMSISEDataSource = Earth::InputDataSourceType::SpaceWeatherFile;
            
            return std::make_unique<NRLMSISE00Impl>(
                aType,
                anInputDataSourceType,
                NRLMSISE00::InputDataSourceType::SpaceWeatherFile,
                anEarthFrameSPtr,
                anEarthRadius,
                anEarthFlattening,
                aSunCelestialSPtr
            );
```

Maybe we could do something like this instead of throwing an error here. Using the space weather file seems like a reasonable default to me and matches current behavior.

Reply:
I like that suggestion! thanks!

Reply:
Applied. Default for NRLMSISE and Undefined will be CSSISpaceWeatherFile

Reply:
Even better, with the default values in python, no breaking changes and combined with your suggestion here:
- Exponential > Undefined. all good
- Exponential > Other. all good
- NRLMSISE > Undefined > meaning CSSI. All good. We can keep the same behavior we had before
- NRLMSISE > CSSI. All good
- NRLMSISE > Constant. Then all good